cli/src/main/java/org/jboss/as/cli/CommandLineMain.java

```diff
@@ -151,4 +158,29 @@ private static Properties createLogManagerConfig(final String level) {
         properties.setProperty("formatter.PATTERN.properties", "pattern");
         return properties;
     }
+
+    /**
+     * Creates a file marker under $JBOSS_HOME/bin directory. This is used by the shutdown operation to understand from where we have
+     * launched CLI instance when we are performing an update of a remote installation.
+     */
+    private static void createClientMarker() {
+        try {
+            final String jbossHome = WildFlySecurityManager.getEnvPropertyPrivileged("JBOSS_HOME", null);
+            if (jbossHome != null) {
+                final Path cliMarkerPath = Paths.get(jbossHome).resolve("bin").resolve(Util.CLI_MARKER);
+                Files.deleteIfExists(cliMarkerPath);
+                Files.createFile(cliMarkerPath);
+
+                final File cliMarkerFile = cliMarkerPath.toFile();
+                cliMarkerFile.deleteOnExit();
+                final String data = String.valueOf(System.currentTimeMillis());
+                try (FileWriter writer = new FileWriter(cliMarkerFile, true)) {
+                    writer.write(data);
+                }
+                WildFlySecurityManager.setPropertyPrivileged(Util.CLI_MARKER_VALUE, data);
+            }
+        } catch (Throwable e) {
+            // explicitly ignore this.
```

Comment:
Could we have a warning displayed. I imagine the case where part of the JBOSS_HOME sub directories would be readonly.

Reply:
Yes, this is explicitly ignored because of the read-only use case (or any other exception).

We could log a WARN (we should move the method invocation after the one that setups the logging) but I considered it as a corner case that could go silently; if that occurs the logic will assume we are using a local CLI instance and it will get closed when the update is in place just because we won't be able to detect whether it is local.

We can add a WARN, but, shouldn't it be just noise since it is a very particular case only related to the shutdown when performing an installation? WDYT?

Reply:
Or at the very least is any logging possible?  Catch blocks that drop the Throwable somehow end up being the place you need more information from when remote debugging an issue.